### DISCUSSION TITLE
Running MD simulations using a trained model with LAMMPS

### DISCUSSION THREAD

**Discussion by omidshy** [deepmodeling/deepmd-kit#2270] [Q&A]
Hi,

when I run two identical MD simulations using the same DP model with LAMMPS (starting from the same LAMMPS restart file), the resulting simulations (trajectories) are not the same.

Is this an expected behavior from a trained DP model?

**Comment by njzjz** [comment]
Did you reset the velocity? According to [LAMMPS documentation](https://docs.lammps.org/velocity.html)

> If the [velocity](https://docs.lammps.org/velocity.html) command is used to set initial atom velocities, a particular atom can be assigned a different velocity when the problem is run on a different number of processors or on different machines. If this happens, the phase space trajectories of the two simulations will rapidly diverge. See the discussion of the loop option in the [velocity](https://docs.lammps.org/velocity.html) command for details and options that avoid this issue.

**Reply by omidshy** [reply]
Since I do not use a ``velocity`` command in the restart runs, I assume that velocities are read from the restart file.
The atomic positions and velocities from both simulations are identical initially (for a few thousand steps), but slowly diverge as the simulations continue.

**Reply by njzjz** [reply]
> Since I do not use a `velocity` command in the restart runs, I assume that velocities are read from the restart file. The atomic positions and velocities from both simulations are identical initially (for a few thousand steps), but slowly diverge as the simulations continue.

You may meet the same case as #1656. If so, this is not a real problem. The precision float64 only gives 15 accurate digits, so it is expected to see cumulative errors.
Which hardware do you use? If you use GPUs, you can see [NVIDIA's slides](https://developer.download.nvidia.com/video/gputechconf/gtc/2019/presentation/s9911-determinism-in-deep-learning.pdf) about determinism.

**Reply by omidshy** [reply]
The plots of atomic coordinates from my simulations are similar to the reported plots in [#1656](https://github.com/deepmodeling/deepmd-kit/issues/1656).
I use NVIDIA GPUs. The mentioned NVIDIA slides discuss some solutions to the "Non-Determinism" problem. I am curious if these problems are addressed already.

**Reply by njzjz** [reply]
> I am curious if these problems are addressed already.

See [NVIDIA/framework-determinism](https://github.com/NVIDIA/framework-determinism) repository.
